### PR TITLE
Fix /frontend container build issues

### PR DESCRIPTION
**Pull request by mjlshen**
There were specific issues building the /frontend container when using colima on macOS that this PR fixes with documentation. However, /frontend was also failing to build due to go module/workspace issues the dependabot is also running into: https://github.com/Azure/ARO-HCP/security/dependabot/6. This PR fixes our build issues and also shifts to `runtime/debug.buildInfo` so that we don't need to leverage a build-time ldflag `Version` variable anymore.
 
### What this PR does
* add git into Dockerfile build step
* remove static Version variable in favor of buildinfo
* replace statements are needed to correctly import /internal into /frontend
* store the commit hash via runtime/debug.buildInfo instead of ldflags
* describe the required colima configuration for macOS

e.g. Sample current output:
```bash
❯ ARO_HCP_IMAGE_ACR=mshen make frontend-build-container

docker build --platform="linux/amd64" -f "frontend/Dockerfile" -t mshen.azurecr.io/arohcpfrontend:50eacd9 .
[+] Building 0.4s (12/12) FINISHED                                                                                                                                                                 ...

❯ docker run --rm -it mshen.azurecr.io/arohcpfrontend:50eacd9                                    
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
{"time":"2024-04-29T23:51:02.09201068Z","level":"INFO","msg":"ARO HCP Frontend (50eacd9168e28c817f57154aec9e2ca28b6e2859) started"}
{"time":"2024-04-29T23:51:02.099208117Z","level":"INFO","msg":"listening on 0.0.0.0:8443"}
^C{"time":"2024-04-29T23:51:03.684648398Z","level":"INFO","msg":"caught interrupt signal"}
{"time":"2024-04-29T23:51:03.686885321Z","level":"INFO","msg":"ARO HCP Frontend (50eacd9168e28c817f57154aec9e2ca28b6e2859) stopped"}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ brew install colima
 
 Before running your devcontainer, make sure colima is started.
 ```bash
-colima start
+colima start --cpu 4 --memory 8 --vz-rosetta --vm-type=vz
```
Comment (mjlshen) [inline README.md:41]: fyi @s-amann this alone should largely remediate macOS build issues.

File: frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Base and builder image will need to be replaced by Fips compliant one
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.9-1161 AS builder
+RUN microdnf install -y tar make git
```
Comment (mjlshen) [inline frontend/Dockerfile:3]: Notably switched the builder container to ubi-minimal as well and am installing git into the builder container

File: frontend/Makefile [hunk 3]
```diff
@@ -1,15 +1,8 @@
...
-	VERSION = $(TAG)
-endif
 
+.PHONY: frontend clean
 frontend:
-	go build -ldflags "-X main.Version=$(VERSION)" -o aro-hcp-frontend .
+	go build -o aro-hcp-frontend .
 
 clean:
-	rm --force aro-hcp-frontend
```
Comment (mjlshen) [inline frontend/Makefile:15]: The `--force` flag doesn't exist on macOS (by default anyway...)

File: Makefile [hunk 4]
```diff
@@ -27,10 +19,10 @@ lint:
 
 .PHONY: frontend
 frontend:
-	go build -ldflags "-X main.Version=$(VERSION)" -o aro-hcp-frontend ./frontend
```
Comment (mbarnes) [inline Makefile:30]: I'm fine with removing `VERSION` but can you replace the `Version` variable in `frontend/main.go`?  Maybe pass the commit hash like `-ldflags "-X main.Commit=$(COMMIT)".  This is just so _some_ kind of revision identifier is logged at startup.

**Review by mbarnes — DISMISSED** [review]
Requested one change since we're doing away with versions.

File: Makefile (hunk 4 above)
Comment (mjlshen) [reply Makefile:30]: Oh the revision identifier was somewhat replaced by this chunk in `frontend/main.go` - I'll add this to the PR description

```go
	version := "unknown"
	if info, ok := debug.ReadBuildInfo(); ok {
		for _, setting := range info.Settings {
			if setting.Key == "vcs.revision" {
				version = setting.Value
				break
			}
		}
	}
```

Comment (mbarnes) [reply Makefile:30]: Ok cool, that works for me.

File: README.md (hunk 1 above)
Comment (s-amann) [reply README.md:41]: 👏 thank you

**Review by s-amann — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (s-amann) [reply README.md:41]: if anyone had already ran `colima start` you may need to run `colima stop; colima delete` before attempting to run with these new vm options

**Review by s-amann — APPROVED** [review]

**Pull request merged**